instr.ml

```diff
@@ -322,3 +322,82 @@ let checkpoint_label pc =
 
 let independent instr exp =
   VarSet.is_empty (VarSet.inter (changed_vars instr) (expr_vars exp))
+
+class mapper = object (m)
```

Comment:
I'm ok with `map`, but I don't see the point in `identity_map`: sure, we don't do anything, but the point is to inherit it to define (non-identity) mappers -- so inheritance would immediately break the `identity` part. I think that `generic_map` would also work, but again why do long when we can do short? One approach would have been to define one `map` class for different types (so a mapper on instructions would be parametrized over or inherit a mapper on variables and on expressions), but here the idea is to define one class for all types at once (and we do finer distinctions than the type structure, although this could be fixed by adding more type synonyms).